app/renderer/components/main/clearBrowsingDataPanel.js

```diff
@@ -34,15 +40,37 @@ class ClearBrowsingDataPanel extends React.Component {
     this.onToggleAutocompleteData = this.onToggleSetting.bind(this, 'autocompleteData')
     this.onToggleAutofillData = this.onToggleSetting.bind(this, 'autofillData')
     this.onToggleSavedSiteSettings = this.onToggleSetting.bind(this, 'savedSiteSettings')
+    this.onTogglePublishersClear = this.onToggleSetting.bind(this, 'publishersClear')
+    this.onTogglePaymentHistory = this.onToggleSetting.bind(this, 'paymentHistory')
     this.onClear = this.onClear.bind(this)
     this.onCancel = this.onCancel.bind(this)
+    this.state = {
+      confirmText: false
+    }
   }
 
   onToggleSetting (setting, e) {
     appActions.onToggleBrowsingData(setting, e.target.value)
   }
 
-  onClear () {
+  onClear (check) {
+    if (
```

Comment:
Can we move this conditional to a separate function so it's clearer what's happening?